src/js/permissions_checks.h

```diff
@@ -99,4 +94,47 @@ namespace ccf::js
       }
     }
   }
+  static std::string explain_kv_map_access(
+    ccf::js::KVAccessPermissions permission, ccf::js::TxAccess access)
+  {
+    char const* table_kind = permission == KVAccessPermissions::READ_ONLY ?
+      "read-only" :
+      "inaccessible";
+
+    char const* exec_context = "unknown";
+    switch (access)
+    {
+      case (TxAccess::APP_RW):
+      {
+        exec_context = "application";
+        break;
+      }
+      case (TxAccess::APP_RO):
+      {
+        exec_context = "read-only application";
+        break;
+      }
+      case (TxAccess::GOV_RO):
+      {
+        exec_context = "read-only governance";
+        break;
+      }
+      case (TxAccess::GOV_RW):
+      {
+        exec_context = "read-write governance";
+        break;
+      }
+    }
+
+    static constexpr char const* access_permissions_explanation_url =
```

Comment:
On the one hand, this is a nice thing to have and I am in favour. On the other hand, done as it is, we will have to be careful to remember that if we move/rename this in the future.

Reply:
True. Fwiw, this isn't new, just moved here from `kvhelpers.h`. In the tests for this, we could extract the URL and fetch it to confirm its good, but then we get a flaky test whenever `github.io` is flaky.

As a step towards keeping them in-sync, I've added a comment to the `.rst`, so we're less likely to move/remove that without updating this too.

Reply:
> but then we get a flaky test whenever github.io is flaky

Yes, and a versioning headache too I suspect. I think the check would need to be based on extracting the relative path (strip root of doc prefix) and looking up in git instead.

Anyway, a comment in the .rst is fine I think.